### ISSUE TITLE
scalatest-core_sjs1 for scala 3 brings in the jvm version of scala-xml

### ISSUE THREAD

**Issue by adamw** [scalatest/scalatest#2013]
Scalatest: 3.2.8, Scala: 3.0.0-RC3

From https://github.com/softwaremill/quicklens/issues/67:

running tests using scala.js+scala3 produced a cryptic error `Error during tests:`, while the same build using scala.js+scala2 works correctly. Investigation by @sjrd revealed that the classpath for the tests contains jvm artifacts. Copying from the linked issue:

```
> show quicklensJS3_0_0_RC3/test:fullClasspath
```
reveals that you get the JVM version of scala3-library on the classpath (in addition to the JS one):
```
...
[info] * Attributed(...\Coursier\Cache\v1\https\repo1.maven.org\maven2\org\scala-lang\scala3-library_3.0.0-RC3\3.0.0-RC3\scala3-library_3.0.0-RC3-3.0.0-RC3.jar)
```
In turn,
```
> show quicklensJS3_0_0_RC3/test:dependencyTree
```
reveals that it is brought by the JVM version of scala-xml, which is brought in by the JS version of scalatest-core_sjs1:
```
[info] com.softwaremill.quicklens:quicklens_sjs1_3.0.0-RC3:1.7.1+0-88579318+20210423-1609-SNAPSHOT
[info]   +-org.scala-js:scalajs-library_2.13:1.5.1 [S]
[info]   +-org.scala-js:scalajs-test-bridge_2.13:1.5.1 [S]
[info]   | +-...
[info]   |
[info]   +-org.scala-lang:scala3-library_sjs1_3.0.0-RC3:3.0.0-RC3 [S]
[info]   | +-...
[info]   |
[info]   +-org.scalatest:scalatest-flatspec_sjs1_3.0.0-RC3:3.2.8
[info]   | +-...
[info]   | |
[info]   | +-org.scalatest:scalatest-core_sjs1_3.0.0-RC3:3.2.8
[info]   |   +-org.scala-js:scalajs-library_2.13:1.5.0 (evicted by: 1.5.1)
[info]   |   +-org.scala-js:scalajs-library_2.13:1.5.1 [S]
[info]   |   +-org.scala-js:scalajs-test-interface_2.13:1.5.0 (evicted by: 1.5.1)
[info]   |   +-org.scala-js:scalajs-test-interface_2.13:1.5.1 [S]
[info]   |   | +-org.scala-js:scalajs-library_2.13:1.5.1 [S]
[info]   |   |
[info]   |   +-org.scala-lang.modules:scala-xml_3.0.0-RC3:2.0.0-RC1 // <--- This is bad!
[info]   |   | +-org.scala-lang:scala3-library_3.0.0-RC3:3.0.0-RC3 [S] // <--- which then causes this, which is bad too
[info]   |   |
[info]   |   +-...
[info]   |
[info]   +-...
```

**Comment by bvenners** [comment]
@adamw thanks that's very detailed. I've been wanting to get rid of the XML dependency altogether, but haven't wanted to devote cycles to that as yet given all the other stuff we have had going. It is just used for serialization of events when running remotely. Anyway, we'll investigate and fix the dependency issue in 3.2.9. How urgent a problem would you say this is for you?

**Comment by adamw** [comment]
@bvenners thanks! Not very :) just wanted to release quicklens for scala 3 both jvm & js, so far it's only jvm. It's really a question to quicklens's users, but I suppose this might impact other scala.js + scala 3 projects as well

**Comment by bvenners** [comment]
@adamw Ok. We'll fix it in 3.2.9 which we will release for 3.0.0-final or an RC4 if there is one. It takes several days to do a ScalaTest release now so I'd rather not do another one until the next 3.0.0-RC or the final release.

**Comment by adamw** [comment]
@bvenners the classpath problem is fixed, so I'll close this issue, thanks!
